service/service_construction.go

```diff
@@ -1237,6 +1512,23 @@ func isUnwrapRequest(metadata map[string]interface{}) bool {
 	return false
 }
 
+func isGenericContractCall(metadata map[string]interface{}) bool {
+	if isUnwrap, ok := metadata["bridge_unwrap"]; ok {
+		if unwrapCall, isBool := isUnwrap.(bool); isBool {
+			return !unwrapCall
```

Comment:
Will `unwrapCall` ever be false here? If it is, it seems like we'll return early without checking `method_signature`...

Reply:
I think we only want to return false here if `unwrapCall` is true, otherwise we should continue.

Reply:
for native and token send transaction, there won't be `bridge_unwrap` field in the metadata. In such case, the `ok` will always be false and it will hit the Line 1529 to return false. The `method_signature` length check is being used as a extra safeguard there that if we don't set `bridge_unwrap` but method_signature.length > 0, we still treat it as generic contract call.

Reply:
I guess what I'm saying here is that if you set `bridge_unwrap` to false, it won't actually end up checking the value of any other metadata (with how it is currently written).

Reply:
I see your point, we set `bridge_unwrap=false` only if this is a generic contract call on client side. For native and token send, this field `bridge_unwrap` won't exists.

But I understand it make more sense to not only check `bridge_unwrap` but also the method signature length it it is going to return true.

Will enhance this logic.